pubsub/message.go

```diff
@@ -112,12 +147,37 @@ func (ah *psAckHandler) OnNack() {
 	ah.done(false)
 }
 
+func (ah *psAckHandler) OnAckWithResult() *AckResult {
+	if !ah.exactlyOnceDelivery {
+		return newSuccessAckResult()
+	}
+	ah.ackResult = ipubsub.NewAckResult()
```

Comment:
If this library supports clean shutdown semantics, you might have to store the futures somewhere so you can cleanly close them all at shutdown. Just a thought for the future - this is pt 9 of the "Exactly-once Client Library PR Sequencing" doc.

Reply:
Got it, yeah started keeping track of this now.